crates/miden-objects/src/note/file.rs

```diff
@@ -20,6 +20,7 @@ const MAGIC: &str = "note";
 // ================================================================================================
 
 /// A serialized representation of a note.
+#[derive(Clone, Debug, PartialEq, Eq)]
 pub enum NoteFile {
     /// The note's details aren't known.
     NoteId(NoteId),
```

Comment:
Not related to this PR, but I do wonder whether it was a good idea to include `NoteId` as a variant in this enum. Seems like it is somewhat misleading that `NoteFile` contains just the note ID.

Reply:
Created https://github.com/0xMiden/miden-base/issues/1983 to track this, will merge this in the meantime